x/telemetry/client/cli/query.go

```diff
@@ -90,44 +80,243 @@ func GetQueryCmdExtendedValidators() *cobra.Command {
+func
+GetQueryCmdAvgBlockTime() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:  "avg-block-time [start-date] [end-date]",
+		Args: cobra.MinimumNArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			startDate, endDate, err := ParseDateInterval(args[0], args[1])
+			if err != nil {
+				return sdkerrors.Wrap(err, "failed to parse date interval")
+			}
+
+			queryClient := telemetrytypes.NewQueryClient(clientCtx)
+			res, err := queryClient.AvgBlockTime(cmd.Context(), &telemetrytypes.QueryAvgBlockTimeRequest{
+				StartDate: startDate,
+				EndDate:   endDate,
+			})
+			if err != nil {
+				return sdkerrors.Wrap(err, "failed to query average block time")
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+
+// GetQueryCmdAvgTxFee implements the query parameters command.
+func
+GetQueryCmdAvgTxFee() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:  "avg-tx-fee [start-date] [end-date]",
+		Args: cobra.MinimumNArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			startDate, endDate, err := ParseDateInterval(args[0], args[1])
+			if err != nil {
+				return sdkerrors.Wrap(err, "failed to parse date interval")
+			}
+
+			queryClient := telemetrytypes.NewQueryClient(clientCtx)
+			res, err := queryClient.AvgTxFee(cmd.Context(), &telemetrytypes.QueryAvgTxFeeRequest{
+				StartDate: startDate,
+				EndDate:   endDate,
 			})
+			if err != nil {
+				return sdkerrors.Wrap(err, "failed to query average tx fee")
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+
+// GetQueryCmdTxVolume implements the query parameters command.
+func
+GetQueryCmdTxVolume() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:  "tx-volume [start-date] [end-date]",
+		Args: cobra.MinimumNArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
 			if err != nil {
 				return err
 			}
 
+			startDate, endDate, err := ParseDateInterval(args[0], args[1])
+			if err != nil {
+				return sdkerrors.Wrap(err, "failed to parse date interval")
+			}
+
+			queryClient := telemetrytypes.NewQueryClient(clientCtx)
+			res, err := queryClient.TxVolume(cmd.Context(), &telemetrytypes.QueryTxVolumeRequest{
+				StartDate: startDate,
+				EndDate:   endDate,
+			})
+			if err != nil {
+				return sdkerrors.Wrap(err, "failed to query tx volume")
+			}
+
 			return clientCtx.PrintProto(res)
 		},
 	}
 
 	flags.AddQueryFlagsToCmd(cmd)
 	return cmd
 }
+
+// GetQueryCmdValidatorsBlocks implements the query parameters command.
+func
+GetQueryCmdValidatorsBlocks() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:  "validators-blocks [start-date] [end-date] [limit] [offset] [desc]",
```

Comment:
Let's make pagination params unnecessary and named, like --limit 10, --offset 1 etc. (everywhere)